### PR TITLE
Fixes gas analyser tabletop & storage interactions

### LINKED ISSUE #87354: Gas analyzers scan the tile when placed on a table
### Client Version:

_No response_

### Issue Summary:

it places the gas analyzer on the table alongside scanning the tile it's on

### Round ID:

_No response_

### Test Merge Information:

_No response_

### Reproduction Steps:

click table with gas analyzer

### PR DESCRIPTION
**Pull request by SyncIt21**
## About The Pull Request
- Fixes #87354

The problem is fixed with tables & storage items. Other interactions I'm not aware of let me know.

## Changelog
:cl:
fix: gas analyzer won't scan the floor when placing it on a table
fix: gas analyzer won't scan storage items when placing it inside them
/:cl:

### CODE REVIEW

**Review by MrMelbert — COMMENTED** [review]
I think analyzer should instead be checking `TRAIT_COMBAT_MODE_SKIP_INTERACTION`. 

This fixes the storage items issue.

**Comment by MrMelbert** [comment]
I also really want to avoid passing modifiers to tool_acts, to keep them as simple as possible

**Comment by SyncIt21** [comment]
> I also really want to avoid passing modifiers to tool_acts, to keep them as simple as possible

It's not mandatory to define this arg when overriding tool procs just like the extra args in ui_act() so we can still keep it simple. I needed the modifiers for the tool offset when placed on the table so there's no going around it just for this case.

> TRAIT_COMBAT_MODE_SKIP_INTERACTION

Thanks I'll test this out. Drafting for now

File: code/game/objects/structures/tables_racks.dm [hunk 1]
```diff
@@ -226,6 +226,9 @@
 		deconstruct(TRUE)
 	return ITEM_INTERACT_SUCCESS
 
+/obj/structure/table/analyzer_act(mob/living/user, obj/item/analyzer/tool, list/modifiers)
+	return table_place_act(user, tool, modifiers)
```
Comment (MrMelbert) [inline code/game/objects/structures/tables_racks.dm:230]: We shouldn't need this anymore since tables also have `TRAIT_COMBAT_MODE_SKIP_INTERACTION`

Comment (SyncIt21) [reply code/game/objects/structures/tables_racks.dm:230]: Didn't know that. Yup

**Comment by SyncIt21** [comment]
I have a doubt about `TRAIT_COMBAT_MODE_SKIP_INTERACTION`.

in the docs it specifies "used to skip item interaction IF the user is in combat mode". 
https://github.com/tgstation/tgstation/blob/b0d71024c0c10a0d276ea3119894c27ca7adf0d0/code/__DEFINES/traits/declarations.dm#L1345-L1346

But when looking around i found that items check for this trait in `interact_with_atom()` proc and don't check if the user is in combat mode for e.g. we don't check here for `user.combat_mode`.
https://github.com/tgstation/tgstation/blob/b0d71024c0c10a0d276ea3119894c27ca7adf0d0/code/game/objects/items/devices/laserpointer.dm#L189-L191

 Even in this case for the gas analyzer we don't care if the user is in combat mode and check for this trait. 

Is this a bug on how the trait is checked or can we continue using it in this way?

**Comment by MrMelbert** [comment]
The trait basically just allows items to prefer one interaction if you are on combat mode, and one interaction if you are not

However some things are very combat-agnostic, and you would always want to prefer one interaction over the other, so there wasn't any point in checking for combat mode. 

Ex - laser pointers, you very very rarely (if ever) want to point it at the backpack on your back
Ex - atmos analyzers, you very very rarely (if ever) want to scan the atmos of your backpack

**Pull request merged**